README.md

```diff
@@ -691,6 +691,22 @@ As with any mbed CLI command, `mbed update` uses the current directory as a work
 
 <span class="tips">**Tip: Synchronizing library references:** Before triggering an update, you might want to synchronize any changes that you've made to the program structure by running ``mbed sync``, which will update the necessary library references and get rid of the invalid ones.</span>
 
+## Publishing local program or library
+
+When you create new (local) source control managed program or library, all revision history will exist only locally and the repository wouldn't be associated with remote one. To publish the local repository (while preserving the revision history) please follow these steps:
+1. Create new empty repository on the remote side - either based on public repository hosting service (Github, Bitbucket, mbed.org), service of your own or even different location on your system
+2. Copy the URL/location of the new repository in your clipboard and open command line in the local repository directory (for example change directory to `mbed-os-example/local-lib`).
+3. To associate the local repository:
+ * For Git - run `git remote add origin <url-or-paht-to-your-remote-repo>`
+ * For Mercurial - edit .hg/hgrc and add (or replace if exists):
+ ```
+[paths]
+default = <url-or-paht-to-your-remote-repo>
+```
+4. Run `mbed publish` to publish your changes
+
+In scanario with nested local repositories, start with the leaf repositories first.
```

Comment:
"In a scenario"
"start with the leaf repositories" (you don't need "first" - you already said "start with")